xla/service/gpu/conv_algorithm_picker_test.cc

```diff
@@ -125,5 +140,105 @@ ENTRY main {
   }
 }
 
+TEST_F(GpuConvAlgorithmPickerTest, SetAlgorithmGraphConvF8) {
+  if (!GetCudaComputeCapability().IsAtLeast(
+          se::CudaComputeCapability::HOPPER)) {
+    GTEST_SKIP() << "FP8 convolutions require Hopper or newer architecture.";
+  }
+  constexpr absl::string_view kHlo = R"(
+HloModule module
+apply {
+  a = f32[] parameter(0)
+  b = f32[] parameter(1)
+  ROOT c = f32[] maximum(a, b)
+}
+ENTRY main {
+  input = f8e4m3fn[1,6,6,128] parameter(0)
+  filter = f8e4m3fn[16,3,3,128] parameter(1)
+  input_scale = f32[] parameter(2)
+  input_scale_bcast = f32[1,6,6,128] broadcast(input_scale), dimensions={}
+  filter_scale = f32[] parameter(3)
+  filter_scale_bcast = f32[16,3,3,128] broadcast(filter_scale), dimensions={}
+  input_f32 = f32[1,6,6,128] convert(input)
+  input_unscaled = f32[1,6,6,128] multiply(input_f32, input_scale_bcast)
+  filter_f32 = f32[16,3,3,128] convert(filter)
+  filter_unscaled = f32[16,3,3,128] multiply(filter_f32, filter_scale_bcast)
+  conv_a = f32[1,6,6,16] convolution(input_unscaled, filter_unscaled), window={size=3x3 pad=1_1x1_1}, dim_labels=b01f_o01i->b01f, feature_group_count=1
+  z_scale = f32[] parameter(4)
+  z_scale_bcast = f32[1,6,6,16] broadcast(z_scale), dimensions={}
+  conv_a_scaled = f32[1,6,6,16] multiply(conv_a, z_scale_bcast)
+  c1 = f32[] constant(-448.)
+  c1_bcast = f32[1,6,6,16] broadcast(c1), dimensions={}
+  c2 = f32[] constant(448.)
+  c2_bcast = f32[1,6,6,16] broadcast(c2), dimensions={}
+  conv_a_clamped = f32[1,6,6,16] clamp(c1_bcast, conv_a_scaled, c2_bcast)
+  conv_a_clamped_f8 = f8e4m3fn[1,6,6,16] convert(conv_a_clamped)
+  abs_conv_a = f32[1,6,6,16] abs(conv_a)
+  c0 = f32[] constant(-inf)
+  amax = f32[] reduce(abs_conv_a, c0), dimensions={0,1,2,3}, to_apply=apply
+  ROOT conv_f8 = (f8e4m3fn[1,6,6,16], f32[]) tuple(conv_a_clamped_f8, amax)
+})";
+  TF_ASSERT_OK_AND_ASSIGN(auto m, ParseAndReturnVerifiedModule(kHlo));
+
+  se::Platform* platform = PlatformUtil::GetDefaultPlatform().value();
+  TF_ASSERT_OK_AND_ASSIGN(std::vector<se::StreamExecutor*> executors,
+                          PlatformUtil::GetStreamExecutors(platform));
+  ASSERT_GT(executors.size(), 0);
+  se::StreamExecutor* stream_exec = executors[0];
+
+  const se::GpuComputeCapability& cc = GetCudaComputeCapability();
+  bool changed = false;
+  TF_ASSERT_OK_AND_ASSIGN(changed, RunHloPass(GpuConvRewriter(cc), m.get()));
+  TF_ASSERT_OK_AND_ASSIGN(
+      changed, RunHloPass(CudnnFusedConvRewriter(GetCudaComputeCapability(),
+                                                 GetDnnVersion(), CUDA_VERSION),
+                          m.get()));
+
+  changed = false;
```

Comment:
Should we test the value of change in between those?

Reply:
Done.